tools/github_webhook_processor.php

```diff
@@ -268,7 +281,114 @@ function handle_pr($payload) {
+}
+
+//return the assoc array of login -> balance for prs
+function pr_balances(){
+	$path = pr_balance_json_path();
+	if(file_exists($path))
+		return json_decode(file_get_contents($path), true);
+	else
+		return array();
+}
+
+//returns the difference in PR balance a pull request would cause
+function get_pr_code_friendliness($payload, $oldbalance = null){
+	global $startingPRBalance;
+	if($oldbalance == null)
+		$oldbalance = $startingPRBalance;
+	$labels = get_pr_labels_array($payload);
+	//anything not in this list defaults to 0
+	$label_values = array(
+		'Fix' => 2,
+		'Refactor' => 2,
+		'Code Improvement' => 1,
+		'Priority: High' => 4,
+		'Priority: CRITICAL' => 5,
+		'Atmospherics' => 4,
+		'Logging' => 1,
+		'Feedback' => 1,
+		'Performance' => 3,
+		'Feature' => -1,
+		'Balance/Rebalance' => -1,
+		'Tweak' => -1,
+		'PRB: Reset' => $startingPRBalance - $oldbalance,
+	);
+
+	$affecting = 0;
+	$is_neutral = FALSE;
+	$found_something_positive = false;
+	foreach($labels as $l){
+		if($l == 'PRB: No Update') {	//no effect on balance
+			$affecting = 0;
+			break;
+		}
+		else if(isset($label_values[$l])) {
+			$friendliness = $label_values[$l];
+			if($friendliness > 0)
+				$found_something_positive = true;
+			$affecting = $found_something_positive ? max($affecting, $friendliness) : $friendliness;
+		}
+	}
+	return $affecting;
+}
 
+function is_maintainer($payload, $author){
+	global $maintainer_team_id;
+	$repo_is_org = $payload['pull_request']['base']['repo']['owner']['type'] == 'Organization';
+	if($maintainer_team_id == null || !$repo_is_org) {
+		$collaburl = $payload['pull_request']['base']['repo']['collaborators_url'] . '/' . $author . '/permissions';
+		$perms = json_decode(apisend($collaburl), true);
+		$permlevel = $perms['permission'];
+		return $permlevel == 'admin' || $permlevel == 'write';
+	}
+	else {
+		$check_url = 'https://api.github.com/teams/' . $maintainer_team_id . '/memberships/' . $author;
+		$result = json_decode(apisend($check_url), true);
+		return isset($result['state']);	//this field won't be here if they aren't a member
+	}
+}
+
+//payload is a merged pull request, updates the pr balances file with the correct positive or negative balance based on comments
+function update_pr_balance($payload) {
+	global $startingPRBalance;
+	global $trackPRBalance;
+	if(!$trackPRBalance)
+		return;
+	$author = $payload['pull_request']['user']['login'];
+	if(is_maintainer($payload, $author))	//immune
+		return;
+	$balances = pr_balances();
+	if(!isset($balances[$author]))
+		$balances[$author] = $startingPRBalance;
+	$friendliness = get_pr_code_friendliness($payload, $balances[$author]);
+	$balances[$author] += $friendliness;
+	if($balances[$author] < 0)
+		create_comment($payload, 'Your Fix/Feature pull request delta is currently below zero (' . $balances[$author] . '). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.');
```

Comment:
Make this only show if the merged pr is also negative.

If you get really low, and start doing fix prs to work your way back up, this message showing on merge of positive prs would be lame and unhelpful.

Reply:
On that note, maybe make another message that shows if a author's balance was below 0, but is now above it.